pinot-common/src/main/java/org/apache/pinot/common/config/TableNameBuilder.java

```diff
@@ -132,4 +132,19 @@ public static boolean isOfflineTableResource(String resourceName) {
   public static boolean isRealtimeTableResource(String resourceName) {
     return REALTIME.tableHasTypeSuffix(resourceName);
   }
+
+  /**
+   * ensure that table name ends with type info, if no, create one with the given type
+   * @param tableName the name of the table
+   * @param type the type of the table for it to fill in if the type info is missing
+   * @return the table type name with the type info
+   */
+  public static String ensureTableNameWithType(String tableName, TableType type) {
```

Comment:
where is this used?

Reply:
Please use `TableNameBuilder.forType(type).tableNameWithType(tableName)`

Reply:
removed this method, @kishoreg this method is used in other components that I will submit PR later